### PR TITLE
Log status update without worker

### PR DESCRIPTION
**Pull request by Martchus**
Log status update without worker instead of possibly throwing an exception.

Prevents errors like
```
[Thu Sep  8 21:15:50 2016] [16102:error] Can't call method "get_property" on 
an undefined value at /local/openQA/script/../lib/OpenQA/Schema/Result/Jobs.pm 
line 715.

[Thu Sep  8 21:15:50 2016] [16102:debug] 404 Not Found (0.0217s, 46.083/s)
[Thu Sep  8 21:15:55 2016] [16102:error] Can't call method "get_property" on 
an undefined value at /local/openQA/script/../lib/OpenQA/Schema/Result/Jobs.pm 
line 715.

[Thu Sep  8 21:16:01 2016] [16102:error] Can't call method "get_property" on 
an undefined value at /local/openQA/script/../lib/OpenQA/Schema/Result/Jobs.pm 
line 715.
```

Of course this will not fix the actual problem which might be related to this issue: https://progress.opensuse.org/issues/13526

### CODE REVIEW

**Comment by coolo** [comment]
who would send updates but the worker?

**Comment by Martchus** [comment]
Good question. The answer is what we need for fixing the linked issue.

There was already a check for this condition before. I just added another check around append_log and save_screenshot to get better log messages instead of Perl exception message which don't say what the actual problem is.

**Comment by aaannz** [comment]
What about to check for worker at the beginning and log warning if without worker and return?
This doesn't solve the races, but for that I think we need to use at least optimistic locking and handle db read/write errors.

**Comment by coolo** [comment]
yeah, move the check for worker to the beginning and leave the rest without further if

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -989,13 +989,19 @@ sub update_status {
...
+    }
+
     # that is a bit of an abuse as we don't have anything of the
     # other payload
     if ($status->{uploading}) {
         $self->update({state => UPLOADING});
         return $ret;
     }
 
+
```
Comment (aaannz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: Personal nitpick, I don't like this empty space here. :D

Comment (okurz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: Not just you ;-)

**Pull request merged**